addons/stock/report/report_stock_reception.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import defaultdict
+
+from odoo import _, api, models
+from odoo.tools import float_compare, float_is_zero, format_date
+
+
+class ReceptionReport(models.AbstractModel):
+    _name = 'report.stock.report_reception'
+    _description = "Stock Reception Report"
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        ''' This report is flexibly designed to work with both individual and batch pickings.
+        '''
+        docids = self.env.context.get('default_picking_ids', docids)
+        pickings = self.env['stock.picking'].search([('id', 'in', docids), ('picking_type_code', '!=', 'outgoing'), ('state', '!=', 'cancel')])
+        picking_states = pickings.mapped('state')
+        # unsupported cases
+        if not pickings:
+            msg = "No transfers selected or a delivery order selected"
+        elif 'done' in picking_states and len(set(picking_states)) > 1:
+            pickings = False
+            msg = "This report cannot be used for done and not done transfers at the same time"
+        if not pickings:
+            return {'pickings': False, 'reason': msg}
+
+        # incoming move qtys
+        product_to_qty_draft = defaultdict(float)
+        product_to_qty_to_assign = defaultdict(list)
+        product_to_total_assigned = defaultdict(list)
+
+        # to support batch pickings we need to track the total already assigned
+        assigned_moves = pickings.move_lines.mapped('move_dest_ids')
```

Comment:
You can filterout move_lines where product_id.type != 'product' (and reuse it in the for loop `for move in pickings.move_lines:`)